### PR TITLE
Avoid duped delegations

### PR DESCRIPTION
**Pull request by mllocs**
Fixes https://github.com/coopdevs/decidim-module-action_delegator/issues/26

Ended up moving the `setting` from the DelegationForm to the CreateDelegation command. Form objects don't play well with model attributes and in this particular case I think that the `setting` doesn't belong to the form.

### CODE REVIEW

File: app/views/decidim/action_delegator/admin/delegations/index.html.erb [hunk 1]
```diff
@@ -10,8 +10,8 @@
       <table class="table-list">
         <thead>
           <tr>
-            <th><%= t(".grantee") %></th>
             <th><%= t(".granter") %></th>
+            <th><%= t(".grantee") %></th>
```
Comment (mllocs) [inline app/views/decidim/action_delegator/admin/delegations/index.html.erb:14]: 👀

File: app/controllers/decidim/action_delegator/admin/delegations_controller.rb [hunk 2]
```diff
@@ -69,6 +69,10 @@ def delegation
           @delegation ||= collection.find_by(id: params[:id])
         end
 
+        def delegation_params
+          params.require(:delegation).permit(:granter_id, :grantee_id)
+        end
```
Comment (sauloperez) [inline app/controllers/decidim/action_delegator/admin/delegations_controller.rb:74]: I'm not following. Why not still relying on the form for these two attributes?

File: db/migrate/20200930102243_add_setting_granter_grantee_unique_index_to_delegations.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AddSettingGranterGranteeUniqueIndexToDelegations < ActiveRecord::Migration[5.2]
+  def change
+    add_index :decidim_action_delegator_delegations,
+              [:decidim_action_delegator_setting_id, :granter_id, :grantee_id],
+              unique: true,
+              name: "index_unique_setting_granter_grantee_delegation"
```
Comment (sauloperez) [inline db/migrate/20200930102243_add_setting_granter_grantee_unique_index_to_delegations.rb:8]: We need to also get rid of 

```rb
t.index ["decidim_action_delegator_setting_id"], name: "index_decidim_delegations_on_action_delegator_setting_id"
```

This compound index makes that one useless.

File: spec/forms/decidim/action_delegator/admin/delegation_form_spec.rb [hunk 4]
```diff
@@ -5,8 +5,7 @@
 describe Decidim::ActionDelegator::Admin::DelegationForm do
   subject { described_class.from_params(attributes) }
 
-  let(:attributes) { { granter_id: granter.id, grantee_id: grantee.id, setting: current_setting } }
-  let(:current_setting) { create(:setting) }
+  let(:attributes) { { granter_id: granter.id, grantee_id: grantee.id } }
```
Comment (sauloperez) [inline spec/forms/decidim/action_delegator/admin/delegation_form_spec.rb:8]: it's probably slightly more readable and changeable if we move this let to the first context, where it's used only.

File: db/migrate/20201006084522_remove_setting_id_index_from_delegations.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class RemoveSettingIdIndexFromDelegations < ActiveRecord::Migration[5.2]
+  def change
+    remove_index :accounts, name: "index_decidim_delegations_on_action_delegator_setting_id"
```
Comment (sauloperez) [inline db/migrate/20201006084522_remove_setting_id_index_from_delegations.rb:5]: :warning: wrong table. You meant `decidim_action_delegator_delegations`, right?

**Review by sauloperez — CHANGES_REQUESTED** [review]

File: db/migrate/20201006084522_remove_setting_id_index_from_delegations.rb (hunk 5 above)
Comment (mllocs) [reply db/migrate/20201006084522_remove_setting_id_index_from_delegations.rb:5]: 🤦

**Review by sauloperez — APPROVED** [review]

**Pull request merged**